Fitness model

rneher opened the pull request:
I just realized that you had notes in .gitignore. rename stuff if you feel like. I also merged the clade-frequency branch into this to avoid divergence and conflicts

trvrb committed:
Dynamically calculate clade frequencies.

Clade frequencies come from the same 1 year back selection of viruses.
Frequencies are used to scale branch widths.

Commit:
* introduced .current for tips in the js
 * made function that calculated .diff and sets current based on that
 * .current is distinct from .alive, since .alive is true whenever a
    node is part of the tree given the .current=true tips

Commit:
inherited from object in classes. this provides a bunch of useful methods

Commit:
* cleaned up left-over LBI date variables
  * rewrote tip counting and frequency calculation
  * split dragged into dragged and dragend. the latter updates the plot, the former only the date

Commit:
Merge branch 'master' into fitness_model

Commit:
Merge branch 'fitness_model' of https://github.com/blab/nextflu into fitness_model

Commit:
Added some notes on how estimate frequency trajectories. To be continued

Commit:
added tip counting and frequency calculation for fitness models

Commit:
tip counting and frequency calculation seems to work. introduced minimum
and maxium frequencies for informative clades. should become arguments

Commit:
added comments

trvrb commented:
I'll just rename my local notes/ directory. I agree with merging latest commits from master.

Commit:
added fitting of parameters to model fit. estimates parameters, but the
fit doesn't improve a great deal during the optimization

Commit:
added biopython tree conversion and function to display attributes on
biopython tree. mainly meant for testing. code resurrected from tree_LBI.

Commit:
added transform argument to the calc_LBI function
changed current into alive in analogy to auspice.js

rneher commented:
I worked a little more on the fitness_model. right now, standardized predictors are only calculated for the clades used during model fitting. This is fine for only the fit, but we need them for all nodes at the end when calculating the fitness predictions of the model. the rub is that standardization is season specific. one way out would be to simply calculate all standardized predictors for each season for each node, i.e., temporarily attach a 3 x (number of seasons) matrix to each node which would amount to some 1e6 numbers. computationally that wouldn't be an issue. what do you think?

Commit:
reorganized much of the standardization. now every node as a dict containing
standardized predictors for each season. In addition, I included a affine
fitness contribution that accounts for inaccurate normalization. Normalization
is tricky since it depends on what type of nodes are included with which weight.
